src/HttpClientFactory/Polly/src/DependencyInjection/PollyServiceCollectionExtensions.cs

```diff
@@ -61,5 +61,42 @@ public static IPolicyRegistry<string> AddPolicyRegistry(this IServiceCollection
 
             return registry;
         }
+
+        /// <summary>
+        /// Registers an empty <see cref="PolicyRegistry"/> in the service collection with service types
+        /// <see cref="IPolicyRegistry{String}"/>, and <see cref="IReadOnlyPolicyRegistry{String}"/> and
+        /// uses the specified delegate to configure it.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+        /// <param name="configureRegistry">A delegate that is used to configure an <see cref="IPolicyRegistry{String}"/>.</param>
+        /// <returns>The provided <see cref="IServiceCollection"/>.</returns>
+        public static IServiceCollection AddPolicyRegistry(this IServiceCollection services, Action<IServiceProvider, IPolicyRegistry<string>> configureRegistry)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            if (configureRegistry == null)
+            {
+                throw new ArgumentNullException(nameof(configureRegistry));
+            }
+
+            // Create an empty registry, configure it and register it as an instance.
+            // This is the best way to get a single instance registered using both interfaces.
+            services.AddSingleton(serviceProvider =>
+            {
+                var registry = new PolicyRegistry();
+
+                configureRegistry(serviceProvider, registry);
+
+                return registry;
+            });
+
+            services.AddSingleton<IPolicyRegistry<string>>(serviceProvider => serviceProvider.GetRequiredService<PolicyRegistry>());
+            services.AddSingleton<IReadOnlyPolicyRegistry<string>>(serviceProvider => serviceProvider.GetRequiredService<PolicyRegistry>());
```

Comment:
Do you need to register both? What about ` IConcurrentPolicyRegistry<string>`?

Reply:
The existing `AddPolicyRegistry()` method registers both:

https://github.com/dotnet/aspnetcore/blob/118f41145cb2511a301df630d6504673cdf4512d/src/HttpClientFactory/Polly/src/DependencyInjection/PollyServiceCollectionExtensions.cs#L30-L36

I could update the PR to register the additional interface in that method and this one if you like? Seems to make sense, at least for the `PolicyRegistry<string>` concrete type, but the remaining overload just accepts a `IPolicyRegistry<string>` which might not be of a type implementing the concurrent one. Either it could test the type and register it for that interface too, or not register it and have 2 register 3 interfaces, but the other two?

Reply:
So far `PolicyRegistry` doesn't implement `IConcurrentPolicyRegistry<string>` nor do the other overloads add them. I wouldn't worry about it.